src/microbe_stage/OrganelleDefinition.cs

```diff
@@ -657,6 +657,30 @@ private Vector3 CalculateCenterOffset()
         return offset;
     }
 
+    private LoadedSceneWithModelInfo? TryGetGraphicsForUpgrade(OrganelleUpgrades? upgrades)
+    {
+        if (upgrades != null)
+        {
+            foreach (var availableUpgrade in AvailableUpgrades)
+            {
+                var upgradeGraphics = availableUpgrade.Value.TryGetGraphicsScene();
+
+                if (upgradeGraphics != null)
+                {
+                    if (upgrades.UnlockedFeatures.Contains(availableUpgrade.Key))
```

Comment:
I'd write this the other way around: check if an upgrade is present before doing the logic for fetching its scene.